libtinyaes/aes_eax.c

```diff
@@ -0,0 +1,116 @@
+/*
+ * Phoenix-RTOS
+ *
+ * AES-EAX mode implementation
+ *
+ * Copyright 2020-2021 Phoenix Systems
+ * Author: Gerard Świderski
+ *
+ * %LICENSE%
+ */
+
+#ifdef WITH_AES_EAX
```

Comment:
I think all  functionality ifdefs should be removed (at least for now). Application that uses library will use valid symbols, and rest of them will not be linked into library. 

Until we don't have shared libraries, constraining features is pointless.

Alternatively you should allow to pass enable/disable features from Makefile.

Reply:
I believe some of these functionalities work only for example for a given key length only (currently not being implemented for others) so it might prevent you from enabling eg. AES-256

Regarding the HW acceleration - we might want to build 2 variants of the lib and link desired programs with either sw or hw implementation (to be discussed with @Darchiv as he's probably the most knowledgeable regarding the intended use cases).

Also - we should probably have separate commits for "importing the files as-is" and "fixing the coding style" and maybe "introducing new functionalities" (a little PITA but we would have better history then - and probably easier merging from internal repo if we would ever need it)

Reply:
Or even better: import it with full git history from the internal repo

Reply:
Don't forget to import the unit tests that are in the internal libtinyaes repo, they constitute a whole and which are currently missing (in this PR). Unless there's a plan to move unit-tests to phoenix-rtos-tests ?

Reply:
BTW ITM [kokke/tiny-AES-c](https://github.com/kokke/tiny-AES-c) has some updates since our fork, e.g. [#c3b753](https://github.com/kokke/tiny-AES-c/commit/c3b7532b6deb507c40c0c53cc3c8b49a50766d9b), maybe it's worth to include.